assets/src/dashboard/app/views/templateDetails/karma/templateDetails.karma.js

```diff
@@ -95,7 +95,8 @@ describe('CUJ: Creator can browse templates in grid view: See pre-built template
   }
 
   function getQueryParams() {
-    const params = new URLSearchParams(window.location.search);
+    const arr = window.location.href.split('?');
```

Comment:
Using `window.location.href.split('?')` seems brittle.

I know this is just in Karma tests so it works in practice, but this would fail on something like `wp-admin/edit.php?post_type=web-story&page=stories-dashboard#/templates-gallery/template-detail?id=3&isLocal=false` -- see the two question marks.

We want the search params within the `location.hash` part, not `window.location.search`. So a more robust way of writing would be something like this:

```js
const params = (new URL(window.location.hash, window.location.origin)).searchParams;
return Object.fromEntries(params);
```

Reply:
> see the two question marks.

The split is used BECAUSE of the two question marks. In my tests, this was the url `http://localhost:9876/?id=87748993#/templates-gallery/template-detail?id=10&isLocal=false`. There I passed window.location.href, URLSearchParams only return the first question mark. By using split and the popping the last element off, I always got `id=10&isLocal=false` passed to `URLSearchParams`.

I will give your code snippet a try and get back. But split and getting last one worked everywhere I could think.

Reply:
Sadly your snippet didn't work.

<img width="1312" alt="Screenshot 2021-08-11 at 10 45 55" src="https://user-images.githubusercontent.com/237508/129007998-250ae710-9443-4c93-b398-c7fa153cd3d2.png">

Reply:
Hmm  can you try `window.location.hostname` instead of `window.location.origin`?

Reply:
Didn't work either. 

<img width="433" alt="Screenshot 2021-08-11 at 11 08 41" src="https://user-images.githubusercontent.com/237508/129011254-246a63d1-4c4e-43ca-b53e-08ac7c476917.png">

Reply:
Ah, of course `host` doesn't have a protocol. Since the base URL is really irrelevant, could just as well hardcode this:


```js
const params = (new URL(window.location.hash, 'http://localhost')).searchParams;
return Object.fromEntries(params);
```

I understand that the `pop()` method works and is clever at that, but it's not immediately obvious from looking at the code what it does. So it's a little hard to grasp.

I know I'm getting nit-picky here but I think using `window.location.hash` like above would make the intention much more clear.

Reply:
> I know I'm getting nit-picky here but I think using window.location.hash like above would make the intention much more clear.

No I agree, I was going to put a comment to explain it. The code before was also super hard for me to understand. 

> const params = (new URL(window.location.hash, 'http://localhost')).searchParams;

I tried this and it didn't work either. I think the issue, is that query string is part of the hash value. These url functions just think it is part of the hash. You have to extract the query from the hash value to work with it. If you want to do some testing, go to `http://localhost:8899/?id=87748993#/templates-gallery/template-detail?id=10&isLocal=false` locally. The page is nothing, but it is the right url structure and can play around with this in the console.

Reply:
The issue is that the `#` is part of `window.location.hash`.

So this works:

```js
const params = (new URL('/templates-gallery/template-detail?id=10&isLocal=false', 'http://localhost')).searchParams;
console.log(params.get('id')) // 10
```

But this doesn't:

```js
const params = (new URL('#/templates-gallery/template-detail?id=10&isLocal=false', 'http://localhost')).searchParams;
console.log(params.get('id')) // null
```

To remove the hash, could do `window.location.hash.substr(1)`:

```js
const params = (new URL(window.location.hash.substr(1), 'http://localhost')).searchParams;
return Object.fromEntries(params);
```

Reply:
>is that query string is part of the hash value 

Yeah, that is what I thought. 

I am going to put this out there, this isn't super readable to me. 

> const params = (new URL(window.location.hash.substr(1), 'http://localhost')).searchParams;

If the point is to improve readability, then I think what is currently there is better.

Reply:
Just coming back to the original snippet:

```js
// http://localhost:8899/?id=87748993#/templates-gallery/template-detail?id=10&isLocal=false -> { id: 10, isLocal: false }
const params = (new URL(window.location.hash.substr(1), window.location.origin)).searchParams;
return Object.fromEntries(params);
```

That doesn't look too bad IMO

Reply:
> const params = (new URL(window.location.hash.substr(1), 'http://localhost')).searchParams;

This also didn't work. 
![Screenshot 2021-08-11 at 13 40 55](https://user-images.githubusercontent.com/237508/129030362-f49ebf30-eb7e-4de1-bf25-07a703a338d8.png)

Reply:
Yeah but have you tried my latest suggestion?

Reply:
> Yeah but have you tried my latest suggestion?

Yes :(

<img width="646" alt="Screenshot 2021-08-11 at 23 22 02" src="https://user-images.githubusercontent.com/237508/129111647-28b6dd6e-e2fe-4778-8f79-0cb0eb3471f8.png">

Reply:
Works!